modules/friendlyshared.js

```diff
@@ -16,15 +16,17 @@
 Twinkle.shared = function friendlyshared() {
 	if( mw.config.get('wgNamespaceNumber') === 3 && Morebits.isIPAddress(mw.config.get('wgTitle')) ) {
 		var username = mw.config.get('wgTitle').split( '/' )[0].replace( /\"/, "\\\""); // only first part before any slashes
-		Twinkle.addPortletLink( function(){ Twinkle.shared.callback(username); }, "Shared IP", "friendly-shared", "Shared IP tagging" );
+		Twinkle.addPortletLink( function(){ Twinkle.shared.callback(username); }, $.i18n._('tw-sharedip-label'), "friendly-shared", 'tw-sharedip-label-hover' );
```

Comment:
The bigger picture looks mostly OK, which is why I didn't comment on it. Ideally we'd get rid of all the enwiki-specific logic, but it's best to go partway to begin with and get it right, I reckon.

I guess my only concern is still the separation into i18n and l10n. I think we need to remember that different wikis in the same language might have different names for the same process, so the distinction between i18n and l10n is quite blurry.